lib/src/components/NxTable/__tests__/NxTable.test.tsx

```diff
@@ -85,82 +91,369 @@ describe('NxTable', function() {
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('true');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow isFilterHeader={false}>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow isFilterHeader={null}>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+
+      rerender(
+        <NxTable>
+          <NxTableBody>
+            <NxTableRow>
+              <NxTableCell>Foo</NxTableCell>
+              <NxTableCell>Bar</NxTableCell>
+              <NxTableCell />
+              <NxTableCell>Baz</NxTableCell>
+              <NxTableCell><ContextDependentChild /></NxTableCell>
+            </NxTableRow>
+          </NxTableBody>
+        </NxTable>
+      );
+
+      expect(screen.getByTestId('context-dependent')).toHaveTextContent('false');
+    });
+  });
+
+  describe('NxTableCell', function() {
+    const renderHeaderComponent = (extraProps?: NxTableCellProps) => render(
+      <NxTable>
+        <NxTableHead>
+          <NxTableRow>
+            <HeaderContext.Provider value={true}>
+              <NxTableCell { ...extraProps }/>
+            </HeaderContext.Provider>
+          </NxTableRow>
+        </NxTableHead>
+      </NxTable>
+    ).getByRole('columnheader') as HTMLElement;
+
+    describe('when the chevron prop is true', function() {
+      describe('when not isHeader', function() {
+        it('sets the button accessible name from the RowContext', function() {
+          render(
+            <NxTable>
+              <NxTableBody>
+                <NxTableRow>
+                  <RowContext.Provider value={{ label: 'foobar', isFilterHeader: false }}>
+                    <NxTableCell chevron/>
+                  </RowContext.Provider>
+                </NxTableRow>
+              </NxTableBody>
+            </NxTable>
+          );
+
+          expect(screen.getByRole('button')).toHaveAccessibleName('foobar');
+        });
+      });
+
+      describe('when isHeader', function() {
+        it('renders a visually hidden header cell with the text "Select Row"', function() {
```

Comment:
It being visually hidden isn't really testable here so I'd remove that part of the test name

Reply:
> It being visually hidden isn't really testable here so I'd remove that part of the test name

Renamed here: 87e191edf2126ba712fdb6081718fb505989f131